src/components/rubric-levels-mobile.js

```diff
@@ -0,0 +1,1038 @@
+
+	firstUpdated() {
+		this._handleResize();
+		new ResizeObserver(this._handleResize.bind(this)).observe(this.shadowRoot.host);
+	}
+
+	render() {
+		if (this._width === 0) {
+			return;
+		}
+		return html`
+			<div
+				role="button"
+				class="${this._getLeftIteratorClass(this._selected, this._focused, this._levelEntities)}"
+				@mousedown="${this._handleLeft}"
+			>
+				<div class="level-iterator">
+					<d2l-icon icon="d2l-tier1:chevron-left"></d2l-icon>
+				</div>
+			</div>
+			<div class="levels-container" @scroll="${this._handleScroll}" @mousemove="${this._handleDrag}">
+				${this._renderGradient()}
+				${this._renderThumb()}
+				<d2l-offscreen id="${this._getCriterionId()}">
+					${this.name}
+				</d2l-offscreen>
+				<div
+					class="levels"
+					role="radiogroup"
+					aria-labelledby="${this._getCriterionId()}"
+				>
+					${this._levelEntities.map((level, index) => this._renderLevel(level, index))}
+				</div>
+			</div>
+			<div
+				role="button"
+				class="${this._getRightIteratorClass(this._selected, this._focused, this._levelEntities)}"
+				@mousedown="${this._handleRight}"
+			>
+				<div class="level-iterator">
+					<d2l-icon icon="d2l-tier1:chevron-right"></d2l-icon>
+				</div>
+			</div>
+			<d2l-offscreen class="criterion-status">
+				<p role="status" aria-live="assertive"></p>
+			</d2l-offscreen>
+		`;
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('_levelEntities') || changedProperties.has('_selected') || changedProperties.has('_focused') || changedProperties.has('_width')) {
+			this._updateScroll(this._levelEntities, this._selected, this._focused, this._width);
+		}
+		if (changedProperties.has('_selected') || changedProperties.has('_hovered') || changedProperties.has('_focused')) {
+			this.dispatchEvent(new CustomEvent('d2l-rubric-level-visible-change', {
+				detail: {
+					levelVisible: this._getVisibleLevel(this._selected, this._hovered, this._focused)
+				},
+				bubbles: true,
+				composed: true,
+			}));
+		}
+		if ((changedProperties.has('criterionCells') || changedProperties.has('_assessmentEntity') || changedProperties.has('_width')) && this.criterionCells && this._assessmentEntity && this._width > 0) {
+			this._selectAssessedLevel(this.criterionCells, this._assessmentEntity);
+		}
+	}
+
+	deselect() {
+		if (this._selected === -1) {
+			return;
+		}
+		const radio = this._getRadioAt(this._selected);
+		if (!radio) {
+			return;
+		}
+		this._queueAction(radio.attributes.cellHref.value, true);
+	}
+
+	focus() {
+		const index = this._selected === -1 ? 0 : this._selected;
+		const radio = this._getRadioAt(index);
+		if (radio) {
+			radio.focus();
+		}
+	}
+
+	getSelfLink(entity) {
+		return entity?.getLinkByRel('self')?.href || '';
+	}
+
+	_canScrollLeft(scroll) {
+		return scroll > 0;
+	}
+
+	_canScrollRight(scroll, visibleWidth) {
+		const scrollContainer = this.shadowRoot && this.shadowRoot.querySelector('.levels-container');
```

Comment:
```suggestion
		const scrollContainer = this.shadowRoot?.querySelector('.levels-container');
```
I'm not sure what scenario could have the shadow root resolve as nullish. Doesn't hurt to be sure I guess